### ISSUE TITLE
Nodes do not have to be mutually linked?

### ISSUE THREAD

**Issue by Josef-Vonasek** [MaiaVictor/abstract-algorithm#13]
zero.lam = `0`
```
PS E:\abstract-algorithm> node .\src\main.js --bruijn --dump .\lib\zero.lam
{ loops: 9, rules: 0, betas: 0, dupls: 0, annis: 0 }
0 '1:0 0:1 0:2 0:0'
1 '0:0 2:0 3:0 1:3'
2 '1:1 2:2 2:1 0:3'
3 '1:2 3:2 3:1 1:3'
4 '3:1 4:2 4:1 0:0'
##a
```

4th node points to 3rd but 3rd does not point to 4th. I thought that all nodes must be linked mutually. 

Btw, what is **era**?

**Comment by MaiaVictor** [comment]
If I understand this graph from just looking (I've not drawn it), 3rd is actually garbage produced from the `toNet()` function. It is an erase (`era`) node. When `toNet()` creates a lambda node, it also creates an erase node (a node with kind 0, and with its left-right port pointing to each-other), and links it to the variable port (1) of that lambda node. When there is no occurrence of the lambda's variable, that node will point to the erased part of the graph (but it will never actually be reduced, because of the lazy evaluation; that's why we don't need different reduction rules for erasure). When there is one occurrence of that lambda's variable, that `era` node is detached (lines 102-104) and the variable port of the lambda node is connected to it. If there are more occurrences of that lambda's variable, duplication nodes are introduced (lines 107-113).

So, in short, it is garbage produced by `toNet()`. It is not connected to the graph and does not affect its evaluation. `toNet()` could probably be improved slightly to avoid producing that garbage.
